fastlane_core/lib/fastlane_core/configuration/configuration.rb

```diff
@@ -185,33 +205,22 @@ def load_configuration_file(config_file_name = nil, block_for_missing = nil, ski
     def fetch(key, ask: true)
       UI.user_error!("Key '#{key}' must be a symbol. Example :app_id.") unless key.kind_of?(Symbol)
 
-      option = option_for_key(key)
-      UI.user_error!("Could not find option for key :#{key}. Available keys: #{@available_options.collect(&:key).join(', ')}") unless option
+      option = verify_options_key!(key)
 
-      value = @values[key]
+      # Same order as https://docs.fastlane.tools/advanced/#priorities-of-parameters-and-options
+      value = if @values.key? key and !@values[key].nil?
+                @values[key]
+              elsif option.env_name and !ENV[option.env_name].nil?
+                ENV[option.env_name].dup
+              elsif self.config_file_options.key? key
+                self.config_file_options[key]
+              else
+                option.default_value
+              end
 
       value = option.auto_convert_value(value)
-
-      # `if value == nil` instead of ||= because false is also a valid value
-      if value.nil? and option.env_name and ENV[option.env_name]
-
-        # We want to inform the user that we took the value
-        # from an environment variable
-        # however we don't print the actual value, as it may contain sensitive information
-        # The user can easily find the actual value by print out the environment
-        UI.verbose("Taking value for '#{key}' from environment variable '#{option.env_name}'")
-
-        value = option.auto_convert_value(ENV[option.env_name].dup)
-        option.verify!(value) if value
-      end
-
-      value = option.default_value if value.nil?
       value = nil if value.nil? and !option.string? # by default boolean flags are false
```

Comment:
I have no idea what this line is supposed to do so I kept it in

Reply:
I think the answer is it does nothing, since it says, effectively, `value = nil if value == nil`. I wonder if the intention was `value = false if value.nil? and !option.string?`. Since Ruby has no Boolean type, you cannot specify `type: Boolean` for these flags, but usually specify `is_string: false`. So if the end result is nil, it would set them to false. As long as they're not strings. But this would then also happen for things like Array params (false if none specified), which seems dubious. You wouldn't be able to do `if params[:array_param].nil?` in an action if it were false instead of nil.